contracts/Core/BondManager.sol

```diff
@@ -0,0 +1,229 @@
+
+        databonds[bondId].bond = databonds[bondId].bond - bond;
+        bondLocks[bondId][msg.sender] = Structs.Lock(bond, epoch + withdrawLockPeriod);
+
+        if (databonds[bondId].bond < minBond && databonds[bondId].active) {
+            for (uint8 i = 0; i < databondCollections.length; i++) {
+                if (databondCollections[i] == databonds[bondId].collectionId) {
+                    databondCollections[i] = databondCollections[databondCollections.length - 1];
+                    // slither-disable-next-line costly-loop
+                    databondCollections.pop();
+                    break;
+                }
+            }
+            databonds[bondId].active = false;
+            collectionManager.setCollectionStatus(false, databonds[bondId].collectionId);
+        }
+    }
+
+    function withdrawBond(uint32 bondId) external databondCreatorCheck(bondId, msg.sender) {
+        uint32 epoch = getEpoch();
+        require(bondLocks[bondId][msg.sender].amount != 0, "no lock created");
+        // slither-disable-next-line timestamp
+        require(bondLocks[bondId][msg.sender].unlockAfter <= epoch, "Withdraw epoch not reached");
+
+        uint256 withdrawAmount = bondLocks[bondId][msg.sender].amount;
+        bondLocks[bondId][msg.sender] = Structs.Lock(0, 0);
+
+        require(razor.transfer(msg.sender, withdrawAmount), "couldnt transfer");
+    }
+
+    function setDatabondStatus(bool databondStatus, uint16 bondId) external databondCreatorCheck(bondId, msg.sender) {
+        uint32 epoch = getEpoch();
+        require(databondStatus != databonds[bondId].active, "status not being changed");
+        require(databonds[bondId].bond >= minBond, "bond needs to be >= minbond");
+        // slither-disable-next-line timestamp
+        require(databonds[bondId].epochBondLastUpdated + epochLimitForUpdateBond <= epoch, "databond been updated recently");
+        if (databondStatus) {
+            databondCollections.push(databonds[bondId].collectionId);
+        } else {
+            for (uint8 i = 0; i < databondCollections.length; i++) {
+                if (databondCollections[i] == databonds[bondId].collectionId) {
+                    databondCollections[i] = databondCollections[databondCollections.length - 1];
+                    // slither-disable-next-line costly-loop
+                    databondCollections.pop();
+                    break;
+                }
+            }
+        }
+        databonds[bondId].active = databondStatus;
+        databonds[bondId].epochBondLastUpdated = getEpoch();
+        collectionManager.setCollectionStatus(databondStatus, databonds[bondId].collectionId);
+    }
+
+    function setOccurrence() external override onlyRole(OCCURRENCE_CALCULATOR_ROLE) {
+        for (uint32 i = 1; i <= numDataBond; i++) {
+            if (databonds[i].active && databonds[i].bond >= minBond) {
+                uint256 occurrence = (databonds[i].jobIds.length * depositPerJob) / databonds[i].bond;
```

Comment:
customisable bond not taken into consideration anymore here. check for desired occurrence is greater than permissible occurrence